src/AspNetCoreController/Example/Startup.cs

```diff
@@ -0,0 +1,57 @@
+using GraphQL;
+using GraphQL.MicrosoftDI;
+using GraphQL.SystemTextJson;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using StarWars;
+
+namespace Example
+{
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddGraphQL()
+                .AddSchema<StarWarsSchema>()
+                .AddSystemTextJson()
+                .AddValidationRule<InputValidationRule>()
+                .AddGraphTypes(typeof(StarWarsSchema).Assembly)
+                .AddMetrics(_ => true, (provider, _) => provider.GetRequiredService<IOptions<GraphQLSettings>>().Value.EnableMetrics);
+
+            services.Configure<GraphQLSettings>(Configuration.GetSection("GraphQLSettings"));
+            services.AddSingleton<StarWarsData>();
+            services.AddLogging(builder => builder.AddConsole());
+            services.AddHttpContextAccessor();
+            services.AddControllersWithViews();
+        }
+
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            if (env.IsDevelopment())
+                app.UseDeveloperExceptionPage();
+
+            app.UseRouting();
```

Comment:
How does it work without `AddRouting` ? Does `AddControllersWithViews` contain `AddRouting` ?

Reply:
I don't know; I guess so.  UseEndpoints however threw an exception and said that UseRouting was required, so I added that.